python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1115,10 +1116,10 @@ def create_reexecuted_run(
         external_pipeline: "ExternalPipeline",
         strategy: "ReexecutionStrategy",
         extra_tags: Optional[Dict[str, Any]] = None,
-        run_config: Optional[Dict[str, Any]] = None,
+        run_config: Optional[Mapping[str, Any]] = None,
         mode: Optional[str] = None,
         use_parent_run_tags: bool = False,
-    ) -> DagsterRun:
+    ) -> PipelineRun:
```

Comment:
why?

Reply:
mypy was complaining: `DagsterRun` extends `PipelineRun`, so not all `PipelineRuns` are necessarily `DagsterRun`s. The return type of `create_run` is `PipelineRun`. Fixing that would have required deeper changes than I wanted to make in this PR.